project/goruntime.go

```diff
@@ -8,26 +8,27 @@ import (
 	"github.com/moby/buildkit/client/llb"
 	bkgw "github.com/moby/buildkit/frontend/gateway/client"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
-	"go.dagger.io/dagger/core/filesystem"
+	"go.dagger.io/dagger/core"
 )
 
-func (p *State) goRuntime(ctx context.Context, subpath string, gw bkgw.Client, platform specs.Platform) (*filesystem.Filesystem, error) {
-	contextState, err := p.workdir.ToState()
+func (p *State) goRuntime(ctx context.Context, subpath string, gw bkgw.Client, platform specs.Platform) (*core.Directory, error) {
+	contextState, rel, platform, err := p.workdir.Decode()
 	if err != nil {
 		return nil, err
 	}
 	workdir := "/src"
-	return filesystem.FromState(ctx,
+	return core.NewDirectory(ctx,
```

Comment:
Is it possible for the runtimes to return Containers instead so that the image config (env vars and such) are maintained? It's not a huge deal but would be nice to have

Reply:
Oh good catch, I forgot `Filesystem` was closer to `Container`, will look into that now.

Reply:
I looked into this, it's definitely possible. Essentially these direct LLB calls need to turn into `*core.Container` calls. The upside is we'll land on a richer `core.Container` Go API, but the downside is it's a lot of work and it gets a bit verbose with all the extra `err != nil` checks.

One alternative is to stick with LLB and add some kind of `NewContainer` that takes a `llb.State`, but I think we'd need to manually fetch the image config, which doesn't seem great.

Here's the patch so far: https://gist.github.com/vito/65a4c3b513e7f3528cbc4ada8945eec8

Maybe better for a separate PR? Or is this introducing a regression without it?

Reply:
> Maybe better for a separate PR? Or is this introducing a regression without it?

Yeah no need to block this one on it. There has already been some fluctuation in whether extension execution propagates the env or not, so I think most code I've seen handles the case where `PATH` and such isn't set right. Another brief period where it's not propagated should be okay I think.